app/lib/features/chat/pages/room_page.dart

```diff
@@ -42,15 +42,107 @@ class RoomPage extends ConsumerWidget {
+    return AppBar(
+      elevation: 0,
+      automaticallyImplyLeading: inSidebar ? false : true,
+      centerTitle: true,
+      toolbarHeight: 70,
+      flexibleSpace: FrostEffect(
+        child: Container(),
+      ),
+      title: GestureDetector(
+        onTap: () => context.pushNamed(
+          Routes.chatProfile.name,
+          pathParameters: {'roomId': roomId},
+        ),
+        child: Column(
+          mainAxisSize: MainAxisSize.max,
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            Text(
+              roomAvatarInfo.displayName ?? roomId,
+              overflow: TextOverflow.clip,
+              style: Theme.of(context).textTheme.bodyLarge,
+            ),
+            const SizedBox(height: 5),
+            activeMembers.when(
+              data: (members) {
+                int count = members.length;
+                return Text(
+                  L10n.of(context).membersCount(count),
+                  style: Theme.of(context).textTheme.bodySmall,
+                );
+              },
+              skipLoadingOnReload: false,
+              error: (error, stackTrace) => Text(
+                L10n.of(context).errorLoadingMembersCount(error),
+              ),
+              loading: () => Skeletonizer(
+                child: Text(L10n.of(context).membersCount(100)),
+              ),
+            ),
+          ],
+        ),
+      ),
+      actions: [
+        GestureDetector(
+          onTap: () => context.pushNamed(
+            Routes.chatProfile.name,
+            pathParameters: {'roomId': roomId},
+          ),
+          child: Padding(
+            padding: const EdgeInsets.only(right: 10),
+            child: RoomAvatar(
+              roomId: roomId,
+              showParents: true,
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    return ref.watch(chatProvider(roomId)).when(
-          data: (convo) => ChatRoom(convo: convo, inSidebar: inSidebar),
-          error: (e, s) => Center(
-            child: Text(L10n.of(context).loadingRoomFailed(e)),
-          ),
-          loading: () => Center(child: Text(L10n.of(context).loading)),
-        );
+    return OrientationBuilder(
+      builder: (context, orientation) => Scaffold(
+        resizeToAvoidBottomInset: orientation == Orientation.portrait,
+        body: Column(
+          children: [
+            appBar(context, ref),
+            ref.watch(chatProvider(roomId)).when(
+                  data: (convo) => ChatRoom(convo: convo, inSidebar: inSidebar),
+                  error: (e, s) => Center(
+                    child: Text(L10n.of(context).loadingRoomFailed(e)),
+                  ),
+                  loading: () => Center(child: Text(L10n.of(context).loading)),
+                ),
+            chatInput(context, ref),
+          ],
+        ),
+      ),
+    );
+  }
```

Comment:
rendering appbar and chatinput outside of the chatProvider watching as they don't actually need it, they are happy with just the roomId